R/package.R

```diff
@@ -182,15 +182,25 @@ pkgdown_config_path <- function(path) {
   )
 }
 
-read_meta <- function(path) {
+read_meta <- function(path, call = caller_env()) {
   path <- pkgdown_config_path(path)
 
   if (is.null(path)) {
     yaml <- list()
   } else {
-    yaml <- yaml::yaml.load_file(path) %||% list()
+    yaml <- withCallingHandlers(
+      yaml::yaml.load_file(path, error.label = NULL) %||% list(),
+      error = function(e) {
+        cli::cli_abort(c(
+          "x" = "Could not parse the config file.",
+          "i" = "Edit {.path {path}} to fix the probkem."
```

Comment:
```suggestion
          "Could not parse config file at {.path {path}}.",
```

?

Reply:
While we are there, might as well rename this to `read_config()`? https://github.com/r-lib/pkgdown/pull/2556#discussion_r1607123048

Reply:
I think it's ok at `read_meta()` since this the object that becomes `meta` in the pkgdown object.